### PR TITLE
CAN initial version

### PR DESCRIPTION
**Pull request by andreaskuster**
- Added basic CAN functionality
- Added documentation for the other CAN devices

### CODE REVIEW

**Comment by andreaskuster** [comment]
@KarlSchu I am in the process of merging old, not yet merged branches into master. Please have a look and approve at it (since you have re-structured  the main function and know best if this is right).

File: include/definitions.h [hunk 1]
```diff
@@ -47,6 +47,18 @@
 #define SERIAL_TX 15
 #define SERIAL_BAUDRATE 115200
 
+/*
+ * CAN Bus
+ * 
+ *  GPIO25  TX
+ *  GPIO26  RX
+ */
+#define CAN_TX (gpio_num_t)25
```
Comment (KarlSchu) [inline include/definitions.h:56]: Conflicts with #define GPIO_INTERRUPT_PIN 25, should we resolve later...

File: include/definitions.h [hunk 2]
```diff
@@ -47,6 +47,18 @@
...
+#define CAN_RX (gpio_num_t)26
+
+#define CAN_RX_QUEUE 10
+
```
Comment (KarlSchu) [inline include/definitions.h:60]: Missing CAN bus speed definition?

File: src/main.cpp [hunk 3]
```diff
@@ -37,6 +37,8 @@
 #include <Simulator.h>
 #include <Temp.h>
 #include <gpio.h>
+#include <IOExt.h>
```
Comment (KarlSchu) [inline src/main.cpp:40]: What is it used for?

**Comment by KarlSchu** [comment]
We should adjust our formatter settings ;)

File: include/definitions.h (hunk 1 above)
Comment (andreaskuster) [reply include/definitions.h:56]: Ok, let's discuss this on Sunday. I think this already came up once. Didn't you resolve it with @garned?

File: include/definitions.h (hunk 2 above)
Comment (andreaskuster) [reply include/definitions.h:60]: Yes, good idea! I will add it.

File: src/main.cpp (hunk 3 above)
Comment (andreaskuster) [reply src/main.cpp:40]: Shouldn't we at least call `init_ioext` from the main file?

**Comment by andreaskuster** [comment]
> We should adjust our formatter settings ;)

I am actually looking into auto-formatting as a pipeline. Stay tuned :)

File: src/main.cpp (hunk 3 above)
Comment (KarlSchu) [reply src/main.cpp:40]: of course, sorry

**Comment by andreaskuster** [comment]
Autoformating is solved now, see PR #79

**Pull request merged**